bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/rest/api/automowerconnect/AutomowerConnectApi.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.automower.internal.rest.api.automowerconnect;
+
+import org.apache.commons.httpclient.HttpStatus;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.binding.automower.internal.rest.api.HusqvarnaApi;
+import org.openhab.binding.automower.internal.rest.api.automowerconnect.dto.MowerCommandRequest;
+import org.openhab.binding.automower.internal.rest.api.automowerconnect.dto.MowerListResult;
+import org.openhab.binding.automower.internal.rest.api.automowerconnect.dto.MowerResult;
+import org.openhab.binding.automower.internal.rest.exceptions.AutomowerCommunicationException;
+import org.openhab.binding.automower.internal.rest.exceptions.UnauthorizedException;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * Allows access to the AutomowerConnectApi
+ *
+ * @author Markus Pfleger - Initial contribution
+ */
+@NonNullByDefault
+public class AutomowerConnectApi extends HusqvarnaApi {
+
+    public AutomowerConnectApi(HttpClient httpClient) {
+        super(httpClient);
+    }
+
+    @Override
+    public String getBaseUrl() {
+        return "https://api.amc.husqvarna.dev/v1";
+    }
+
+    public MowerListResult getMowers(String appKey, String token) throws AutomowerCommunicationException {
+        final Request request = getHttpClient().newRequest(getBaseUrl() + "/mowers");
+        request.method(HttpMethod.GET);
+
+        request.header("Authorization-Provider", "husqvarna");
+        request.header("Authorization", "Bearer " + token);
+        request.header("X-Api-Key", appKey);
+        request.header("Content-Type", "application/vnd.api+json");
+
+        ContentResponse response;
+        try {
+            response = request.send();
+        } catch (Exception e) {
```

Comment:
This will also catch RuntimeExceptions. Better specify the concrete execeptions. Same for below.
```suggestion
            } catch (InterruptedException | TimeoutException | ExecutionException e) {
```

Reply:
yes